SmartyEngine.php

```diff
@@ -569,13 +570,21 @@ public function getPlugins($extensionName = false)
     public function registerPlugins()
     {
         foreach ($this->getPlugins() as $plugin) {
-            try {
-                $this->smarty->registerPlugin($plugin->getType(), $plugin->getName(), $plugin->getCallback());
-            } catch (\SmartyException $e) {
-                if (null !== $this->logger) {
-                    $this->logger->debug(sprintf("SmartyException caught: %s.", $e->getMessage()));
-                }
-            }
+            $this->registerPlugin($plugin);
+        }
+    }
+
+    /**
+     * register plugin to Smarty.
+     *
+     * @return  void
+     */
+    protected function registerPlugin(PluginInterface $plugin)
+    {
+        // verify that plugin isn't registered yet. That would cause a SmartyException.
+        if (!in_array($plugin->getType() . '_' . $plugin->getName(), $this->registeredPlugins)) {
+            $this->smarty->registerPlugin($plugin->getType(), $plugin->getName(), $plugin->getCallback());
```

Comment:
Would be preferable to "ask" Smarty if the plugin is already registered to avoid keeping state in two places leading to inconsistency potentially. Not sure if Smarty provides this though.